src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -345,6 +338,30 @@ const StudentApplication = ({ userProfile, authentication }) => {
     }
   };
 
+  /**
+   * Callback for changes to off-campus program info
+   * @param {String} offCampusProgramValue The program that the applicant is doing an OC program for
+   * @param {Number} index The index of the applicant in the list
+   */
+  const handleOffCampusInputChange = (offCampusProgramValue, index) => {
+    if (index !== null && index >= 0) {
+      // Get the profile of the selected user
+      let newApplicant = {
+        ...applicants[index],
+        OffCampusProgram: offCampusProgramValue ?? applicants[index].OffCampusProgram,
+      };
+      // Error checking on the hallNameValue before modifying the newHallInfo object
+      setApplicants((prevApplicants) =>
+        // replace the element at index with the new hall info object
+        prevApplicants.map((prevApplicant, j) => (j === index ? newApplicant : prevApplicant)),
+      );
+    } else {
+      setSnackbarText('Something went wrong while trying to add this person. Please try again.');
```

Comment:
This text doesn't seem to correspond to the action that failed. This would be triggered if the user set an off campus program and the update failed, right? Could this error message be more clear?
```suggestion
      setSnackbarText(`Something went wrong while trying to add ${newApplicant.Username}'s off campus program. Please try again.`);
```

Reply:
The `newApplicant` variable would not be declared in this scope, and this is the `else` of the `if` that checks whether the index is valid, so you cannot use `applicants[index].Username`, i.e., there is no valid applicant to put in the message. 
But I agree with what you are getting at, this must be a copy-and-paste that never got updated

Reply:
Suggested alternative: https://github.com/gordon-cs/gordon-360-ui/pull/1082#discussion_r615213719